### ISSUE TITLE
For those interested in running PATH in 2024

### ISSUE THREAD

**Issue by claczny** [KubaWojciechowski/PATH#1]
Dear all,

the README of this tool is very succinct.
Notably, the dependencies have significantly evolved since then and new versions are available.
Also, the way of installing the dependencies might have changed.
I'd like to share how I got the tool running.

# Install the dependencies
I decided to use `conda` for the installation of the dependencies. Note that `pyrosetta` requires actually quite some RAM even during the installation, hence I gave it ca. 16GB of RAM on a cluster.
This is the YAML file I used:
```
name: PATH
channels:
    - https://conda.rosettacommons.org
    - salilab
    - conda-forge
    - bioconda
    - defaults
dependencies:
    - pyrosetta
    - modeller
    - numpy
    - pandas
    - biopython
    - scikit-learn
```
This lead to the installation of the following versions of these tools
```
biopython                 1.83            py312h98912ed_0    conda-forge
pyrosetta                 2024.10+release.2c36cbc         py312_0    https://conda.rosettacommons.org
modeller                  10.5            py312h9bf148f_0    salilab
numpy                     1.26.4          py312heda63a1_0    conda-forge
pandas                    2.2.1           py312hfb8ada1_0    conda-forge
scikit-learn              1.4.1.post1     py312h394d371_0    conda-forge
```

# Getting ready
Note that I had to get a license for `modeller` from https://salilab.org/modeller/registration.html and insert this into `/PATH/TO/MY/CONDA/envs/PATH/lib/modeller-10.5/modlib/modeller/config.py`

I also modified the call to `modeller` in `path1.1.py` to account for the `10.5` version that was installed via `conda`
```
runScript("mod10.5 runMod.py")
```

# Fixing serialization issues
`PATH` loads some pre-computed data via `pickle`. However, there is a fundamental issue using different `scikit-learn` versions for serialized objects.
It seems that `scikit-learn 0.21.2` was used to serialize the models, as this is the version that an `InconsistentVersionWarning` highlights after I applied the following fix.
I added
```
# Taken from https://www.kaggle.com/code/adeepak7/load-old-sklearn-models-with-new-sklearn-package as there was an issue with old-pickled files
import sys
from sklearn.linear_model import _logistic
from sklearn.preprocessing import _data
sys.modules['sklearn.linear_model.logistic'] = _logistic
sys.modules['sklearn.preprocessing.data'] = _data
```
to `path1.1.py` as I was otherwise getting errors due to API changes that seem to have occurred over time.

With this, I was able to eventually run the tool, albeit the `InconsistentVersionWarning` mentions:
```
InconsistentVersionWarning: Trying to unpickle estimator StandardScaler from version 0.21.2 when using version 1.4.1.post1. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
```

In summary, using the approach above, I was able to get the tool running, have it terminate, and produce some output.
Yet, whether the output is valid or not is still unclear.
Maybe this helps though some of you out there.

Best,
Cedric

**Comment by KubaWojciechowski** [comment]
Dear Cedric,
Thank You very much for sharing your solution, I was not aware of these issues. I did some basic testing and it seems like the results are consistent. More tests to come. I think it would be a good idea to add your environment to repo if you don't mind.
Best,
Jakub

**Comment by claczny** [comment]
Dear Jakub,

sure, feel free to add it 👍 
If you want, I can also submit a Pull request.

Best wishes and stay safe,

Cedric

**Comment by KubaWojciechowski** [comment]
Nice, we can do it by pull request so your work will be noticed :)
